src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java

```diff
@@ -578,6 +604,89 @@ private boolean createComponent(Channel channel, User sender, String subcomponen
         return result;
     }
 
+    String createUploadPermissionPR(Channel channel, User sender, String jiraIssue, String forkTo, List<String> ghUsers, List<String> releaseUsers) {
+        String prUrl = "";
+        boolean isPlugin = forkTo.endsWith("-plugin");
+        OutputChannel out = channel.send();
+        if(isPlugin) {
+            String branchName = forkTo + "-permissions";
+            try {
+                if (releaseUsers.size() == 0) {
+                    out.message("No users defined for release permissions, will not create PR");
+                    return null;
+                }
+
+                GitHub github = GitHub.connect();
+                GHOrganization org = github.getOrganization(IrcBotConfig.GITHUB_INFRA_ORGANIZATION);
+                GHRepository repo = org.getRepository(IrcBotConfig.GITHUB_UPLOAD_PERMISSIONS_REPO);
+
+
+                GHContent prTemplate = repo.getFileContent(".github/PULL_REQUEST_TEMPLATE.md");
+                if (prTemplate == null) {
+                    out.message("Could not retrieve PR template for " + IrcBotConfig.GITHUB_UPLOAD_PERMISSIONS_REPO);
+                    return null;
+                }
+
+                String artifactPath = getArtifactPath(github, forkTo);
+                if (StringUtils.isBlank(artifactPath)) {
+                    out.message("Could not resolve artifact path for " + forkTo);
+                    return null;
+                }
+
+                try {
+                    repo.getBranch(branchName);
+                } catch (IOException e) {
```

Comment:
bit weird? can we check if it exists first? I would expect that exception to be thrown pretty much every time?

Reply:
I'll look into that.